src/main/java/pl/senla/hotel/controller/ControllerFacilitySpring.java

```diff
@@ -0,0 +1,122 @@
+
+@RestController
+@RequestMapping("/rooms")
+@Slf4j
+public class ControllerFacilitySpring implements ControllerFacility{
+
+    @Autowired
+    private ServiceFacility serviceFacility;
+
+    @Override
+    @GetMapping
+    public List<RoomDto> readAll() {
+        log.debug("ControllerFacility call ServiceFacility's method 'ReadAll'.");
+        return serviceFacility.readAll().stream()
+                .map(RoomDtoMapperUtil::convertRoomToRoomDto)
+                .toList();
+    }
+
+    @Override
+    @PostMapping("/")
+    public RoomDto create(@RequestBody RoomDto roomDto) throws IllegalAccessException,
+            InvocationTargetException, NoSuchMethodException, InstantiationException {
+        log.debug("ControllerFacility call ServiceFacility's method 'Create'.");
+        return RoomDtoMapperUtil.convertRoomToRoomDto(serviceFacility.create(roomDto));
+    }
+
+    @Override
+    @GetMapping("/{id}")
+    public RoomDto read(@PathVariable Integer id) throws InvocationTargetException, NoSuchMethodException,
+            InstantiationException, IllegalAccessException {
+        log.debug("ControllerFacility call ServiceFacility's method 'Read'.");
+        return RoomDtoMapperUtil.convertRoomToRoomDto(serviceFacility.read(id));
+    }
+
+    @Override
+    @PutMapping("/{id}")
+    public RoomDto update(@PathVariable Integer id, @RequestBody RoomDto RoomDtoNew) throws InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException {
+        log.debug("ControllerFacility call ServiceFacility's method 'Update'.");
+        return RoomDtoMapperUtil.convertRoomToRoomDto(serviceFacility.update(id, RoomDtoNew));
+    }
+
+    @Override
+    @GetMapping("/available/{id}")
+    public RoomDto updateRoomStatusAvailable(@PathVariable Integer id) throws InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException {
+        log.debug("ControllerFacility call ServiceFacility's method 'UpdateRoomStatusAvailable'.");
+        return RoomDtoMapperUtil.convertRoomToRoomDto(serviceFacility.updateRoomStatusAvailable(id));
+    }
+
+    @Override
+    @GetMapping("/repaired/{id}")
+    public RoomDto updateRoomStatusRepaired(@PathVariable Integer id) throws InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException {
+        log.debug("ControllerFacility call ServiceFacility's method 'UpdateRoomStatusRepaired'.");
+        return RoomDtoMapperUtil.convertRoomToRoomDto(serviceFacility.updateRoomStatusRepaired(id));
+    }
+
+    @Override
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Integer id) throws InvocationTargetException, NoSuchMethodException,
+            InstantiationException, IllegalAccessException {
+        log.debug("ControllerFacility call ServiceFacility's method 'Delete'.");
+        serviceFacility.delete(id);
+    }
+
+    @Override
+    @GetMapping("/sort=category")
```

Comment:
better to provide it as query params http://myapp.com/books?sort=author like here and use only one controller method for all sorting

Reply:
I can't understand why query params doesn't work in my app...